force-app/main/default/classes/TestDataFactory.cls

```diff
@@ -137,6 +146,26 @@ public with sharing class TestDataFactory {
         engagement2.NumAbsentServiceDeliveries__c = 0;
         engagements.add(engagement2);
 
+        ProgramEngagement__c engagement3 = new ProgramEngagement__c();
+        engagement3.Name = 'Engagement 3';
+        engagement3.Stage__c = 'Enrolled';
+        engagement3.Contact__c = contact3.Id;
+        engagement3.Program__c = program1.Id;
+        engagement3.Role__c = 'Client';
+        engagement3.NumPresentServiceDeliveries__c = 0;
+        engagement3.NumAbsentServiceDeliveries__c = 0;
+        engagements.add(engagement3);
+
+        ProgramEngagement__c engagement4 = new ProgramEngagement__c();
+        engagement4.Name = 'Engagement 4';
+        engagement4.Stage__c = 'Enrolled';
+        engagement4.Contact__c = contact4.Id;
+        engagement4.Program__c = program2.Id;
+        engagement4.Role__c = 'Client';
+        engagement4.NumPresentServiceDeliveries__c = 0;
```

Comment:
Should these be 0 or should they match what you have in the records below? I think we want to make sure we set them to what we expect and disable the trigger before creating these and the related record?